packages/backend/src/models/AIConfig.ts

```diff
@@ -52,11 +52,25 @@ export function assertString(value: unknown): string {
 export function parseYamlFile(filepath: string, defaultArch: string): AIConfig {
   const raw: string = fs.readFileSync(filepath, 'utf-8');
 
-  const aiLabConfig = jsYaml.load(raw);
-  const application = aiLabConfig?.['application'];
-  if (!application) throw new Error('AIConfig has bad formatting.');
+  const aiLabConfig: unknown = jsYaml.load(raw);
+  if (!aiLabConfig || typeof aiLabConfig !== 'object') {
+    throw new Error('malformed configuration file.');
+  }
 
-  const containers: unknown[] = application['containers'] ?? [];
+  if (!('application' in aiLabConfig)) {
```

Comment:
I think it will help if you introduce a type for the aiLabConfig object

and then do use the type guards

like 

if (!isAiLabConfig(obj)) {
 throw new Error
} 

and you do the check using the https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

Reply:
The `parseYamlFile` will be an all in one, as we already have a type `AIConfig` but it does not match the file, and need to be transformed to fill with default values, map properties to other name (E.g. `gpu-env` -> `gpu_env`) etc.